posthog/models/feature_flag.py

```diff
@@ -75,6 +89,15 @@ def __init__(self, distinct_id: str, feature_flag: FeatureFlag):
     def is_match(self):
         return any(self.is_group_match(group, index) for index, group in enumerate(self.feature_flag.groups))
 
+    def get_matching_variant(self) -> Optional[str]:
+        if not self.is_match():
```

Comment:
Are we required to have this check in this method? 

Or is this just an added safety measure (is this the same check that happens when you call `feature_flag.distinct_id_matches(distinct_id)` upstream?)

Reply:
Good call. No, we don't need it.